Remotion/Data/DomainObjects/Persistence/Rdbms/StorageProviderCommands/MultiDataContainerAssociateWithIDsCommand.cs

```diff
@@ -58,17 +56,29 @@ public ObjectID[] ObjectIDs
       get { return _objectIDs; }
     }
 
-    public IStorageProviderCommand<IEnumerable<DataContainer?>, IRdbmsProviderCommandExecutionContext> Command
+    public IRdbmsProviderReadWriteCommand<IEnumerable<DataContainer?>> Command
     {
       get { return _command; }
     }
 
-    public IEnumerable<ObjectLookupResult<DataContainer>> Execute (IRdbmsProviderCommandExecutionContext executionContext)
+    public IEnumerable<ObjectLookupResult<DataContainer>> Execute (IRdbmsProviderReadWriteCommandExecutionContext executionContext)
```

Comment:
Does not look good because of the type parameters in the interface:
`IEnumerable<ObjectLookupResult<DataContainer>>         IRdbmsProviderCommand<IEnumerable<ObjectLookupResult<DataContainer>>>.Execute (IRdbmsProviderReadWriteCommandExecutionContext executionContext)`

Reply:
ah, I see. we could make an alias-name for this, though. It really depends on: do you want to hide it from the public surface or not. If you want to ide it, we can use a typename-alias via `using IRdbmsProviderCommand= IRdbmsProviderCommand<IEnumerable<ObjectLookupResult<DataContainer>>>` and be done with it :)

Reply:
I think that in a "command with read-only support", both implementations (within a read-write and a read-only context) are of equal rank/importance, so I don't think making on explicitly implement the interface improves understanding/legibility of the code.